lib/patch-series.ts

```diff
@@ -490,7 +490,15 @@ export class PatchSeries {
 
             logger?.log(`\nheader:\n${header}\n**************`);
 
-            mails[i] = header + "\n\nFrom: " + authorMatch[2] + match[2];
+            let body = match[2];
+            if (thisAuthor) {
+                let senderSignoff = `\nSigned-off-by: ${thisAuthor}\n`;
+                let authorSignoff = `\nSigned-off-by: ${authorMatch[2]}\n`;
+                let replaceText = (body.indexOf(authorSignoff) == -1) ? '\n' : authorSignoff;
+                body = body.replace(senderSignoff, replaceText);
+            }
```

Comment:
Could you show an example of what this is trying to solve?

Reply:
If the author already signed-off on the commit, then the signoff inserted by `format-patch` is dropped. If they didn't, the inserted signoff is changed to refer to the author.

Reply:
Wouldn't it be easier to insert that line right away instead of going through `format-patch --signoff` and replacing later?

Reply:
Wouldn't it be easier to insert that line right away instead of going through `format-patch --signoff` and replacing later?